sysext/sysext.go

```diff
@@ -0,0 +1,95 @@
+package sysext
+
+import (
+	"archive/tar"
+	"fmt"
+	"github.com/kairos-io/kairos-sdk/types"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// DefaultAllowListRegex provided for easy use of defaults for confext and sysext
+var DefaultAllowListRegex = regexp.MustCompile(`^usr|^/usr|^etc|^/etc`)
+
+// ExtractFilesFromLastLayer will get an image and a destination and extract the files from the last layer in the image
+// into that destination.
+// It will skip anything that doesn't start with /usr or /etc as its purpose is to get the files for creating a
+// sysextension or a confextension
+// Accepts an allowList in form of regexp.Regexp that will match the files and allow copying
+func ExtractFilesFromLastLayer(image v1.Image, dst string, log types.KairosLogger, allowList *regexp.Regexp) error {
+	layers, _ := image.Layers()
+	numLayers := len(layers)
+	if numLayers == 0 {
+		return extractFilesFromLayer(image, dst, log, allowList, numLayers)
```

Comment:
if `numLayers` is zero, it means there are no layers. Passing `0` to `extractFilesFromLayer` will try to extract the first (zero index) layer but there is none so it will throw "index our of range". No?

Reply:
I dont think so, I think layer 0 is the root layer, so in a dockerfile that its just a "FROM ubuntu" thats the layer 0.

Any other Dockerfile that doesnt at least have the FROM is not valid so I think all containers will have a minimun of 1 layer with index 0.

Reply:
because it can either have 0 or X layers and we want the last, so we cannot do `0 - 1` to get the last layer always, so we do if its 0 just get zero, if its not just get the last one.

In any case, this has changed with @mauromorales suggestion so maybe its clearer now?

Reply:
> so we do if its 0 just get zero, if its not just get the last one.

this is not true. When it's zero (no layers at all), we get index 0, which is the first layer. But we had no layer to begin with.

It's like this simplified code: https://go.dev/play/p/mxai6EnLY4c
You can't access index "0" from a slice with zero elements.

Reply:
ahh I get it now, yes. 

umm. I kind of remember hitting this issue with a 1 layer image. I migth need to check this again.

Reply:
Ok reworked, not sure where I did get the idea of that but yeah, good fucking catch! This was broken for sure, added the fix and added tests that cover that scenario.

Thanks!!